pkg/cmd/integration.go

```diff
@@ -122,51 +123,88 @@ oc plugin mobile create integration <consuming_service_instance_id> <providing_s
 			if err != nil {
-				return err
+				return errors.WithStack(err)
 			}
-			consumerSvcInst, err := bc.scClient.ServicecatalogV1beta1().ServiceInstances(namespace).Get(consumerSvcInstName, meta_v1.GetOptions{})
+			consumerSvcInst, err := bc.scClient.ServicecatalogV1beta1().ServiceInstances(namespace).Get(consumerSvcInstName, metav1.GetOptions{})
 			if err != nil {
-				return err
+				return errors.WithStack(err)
 			}
-			//todo remove the need for these by updating the apbs to read the secrets themselves (https://github.com/feedhenry/keycloak-apb/issues/37)
+			// todo remove the need for these by updating the apbs to read the secrets themselves (https://github.com/feedhenry/keycloak-apb/issues/37)
 			consumerSvc := getService(namespace, consumerSvcInst.Labels["serviceName"], bc.k8Client) // the consumer service
 			providerSvc := getService(namespace, providerSvcInst.Labels["serviceName"], bc.k8Client) // the provider service
 			bindParams := buildBindParams(providerSvc, consumerSvc)
 			objectName := objectName(consumerSvcInstName, providerSvcInstName)
 			binding, err := createBindingObject(consumerSvc.Name, providerSvc.Name, objectName, providerSvcInst.Name, bindParams, objectName)
 			if err != nil {
-				return err
+				return errors.WithStack(err)
 			}
-			if _, err := bc.scClient.ServicecatalogV1beta1().ServiceBindings(namespace).Create(binding); err != nil {
-				return err
+			sb, err := bc.scClient.ServicecatalogV1beta1().ServiceBindings(namespace).Create(binding)
+			if err != nil {
+				return errors.WithStack(err)
 			}
 			preset := podPreset(objectName, objectName, providerSvc.Name, consumerSvc.Name)
 			if _, err := bc.k8Client.SettingsV1alpha1().PodPresets(namespace).Create(preset); err != nil {
 				return errors.Wrap(err, "failed to create pod preset for service ")
 			}
 			redeploy, err := cmd.PersistentFlags().GetBool("auto-redeploy")
 			if err != nil {
-				return err
+				return errors.WithStack(err)
 			}
 			if !redeploy {
 				fmt.Println("you will need to redeploy your service/pod to pick up the changes")
 				return nil
 			}
-			//update the deployment with an annotation
-			dep, err := bc.k8Client.AppsV1beta1().Deployments(namespace).Get(consumerSvc.Name, meta_v1.GetOptions{})
+			// update the deployment with an annotation
+			dep, err := bc.k8Client.AppsV1beta1().Deployments(namespace).Get(consumerSvc.Name, metav1.GetOptions{})
 			if err != nil {
 				return errors.Wrap(err, "failed to get deployment for service "+consumerSvcInstName)
 			}
 			dep.Spec.Template.Labels[providerSvc.Name] = "enabled"
 			if _, err := bc.k8Client.AppsV1beta1().Deployments(namespace).Update(dep); err != nil {
-				return errors.Wrap(err, "failed up update deployment for "+consumerSvcInstName)
+				return errors.Wrap(err, "failed to update deployment for service "+consumerSvcInstName)
+			}
+
+			// exit immediately
+			noWait, err := cmd.PersistentFlags().GetBool("no-wait")
```

Comment:
I think the logic here should go:

```
if noWait && ! redeploy{
  return 
}
//wait logic

if redeploy{
//redeploy logic
}

```

If you want me to assist here  I am happy to do so

Reply:
What happens if a user specifies `no-wait` and `redeploy`? I'm guessing that's not possible so I could just send back the please redeploy message?

Reply:
As mentioned on IRC we have to wait if auto redeploy is set. We will document this in the command docs